Improve osd description with configless mode (#29)
omnivector-solutions/osd-documentation: architecture.rst

================
OSD Architecture
================

The Omnivector SLURM Distribution is built on a suite of automations called
"charms". Charms are the operational components that describe the lifecycle of
a SLURM cluster. A full SLURM deployment comes in the form of multiple charms,
one for each component of SLURM. A "bundle" is a YAML file where multiple
charms can be defined. We use bundles to describe the interconnectivity and
configuration of groups of charms.

> OSD sets up the SLURM cluster in `configless mode

Reviewer: ```suggestion
OSD configures SLURM to operate in `configless mode
```

> <https://slurm.schedmd.com/configless_slurm.html>`_. In this mode, the compute

Reviewer: ```suggestion
<https://slurm.schedmd.com/configless_slurm.html>`_. In this mode, the `slurmctld` process does the work of
distributing the `slurm.conf` file to the nodes running `slurmd`.
```

> and login nodes automatically update their configuration based on the

Reviewer: ```suggestion
```

> controller node.

Reviewer: ```suggestion
```
